spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/security/oauth2/resource/servlet/OAuth2ResourceServerAutoConfigurationTests.java

```diff
@@ -86,6 +134,15 @@ public void autoConfigurationShouldBeConditionalOnJwtAuthenticationTokenClass()
 				.run((context) -> assertThat(getBearerTokenFilter(context)).isNull());
 	}
 
+	@Test
+	public void autoConfigurationShouldBeConditionalOnJwtAuthenticationTokenClassOidcIssuerLocation() {
+		this.contextRunner.withPropertyValues(
+				"spring.security.oauth2.resource.jwt.oidc-issuer-location=http://jwk-oidc-issuer-location.com")
+				.withUserConfiguration(JwtDecoderConfig.class)
+				.withClassLoader(new FilteredClassLoader(JwtAuthenticationToken.class))
+				.run((context) -> assertThat(getBearerTokenFilter(context)).isNull());
+	}
+
```

Comment:
@ayudovin We should probably add a test for what happens when both `oidc-issuer-location` and `jwk-set-uri` are present. My guess is `jwk-set-uri` should win but let's see if @jzheaux thinks otherwise.

Reply:
The other option is to fail hard if both are specified.

Reply:
From my point of view the second option will be better.

Reply:
I think that jwk-set-uri should win because it is more specific.

Reply:
Agreed that `jwk-set-uri` is more specific and should take precedence because of that.

Reply:
ok, now `jwk-set-uri` has precedence.